index.d.ts

```diff
@@ -129,6 +131,41 @@ export namespace Cmd {
   }
   export type PromiseResult<T> = T extends Promise<infer U> ? U : T;
 
+  export function run<Func extends (...args: any[]) => Promise<any> | any>(
+    f: Func,
+    options?: {
```

Comment:
you could avoid repeating the options type by doing something like this.

```ts
type RunFuncType =(...args: any[]) => Promise<any> | any;
export type RunOptions<Func extends RunFuncType, SuccessAction, FailAction> = {
  args?: ArgOrSymbol<Parameters<Func>>;
  failActionCreator: (error: any) => FailAction;
  successActionCreator: (value: PromiseResult<ReturnType<Func>>) => SuccessAction;
  forceSync?: boolean;
  testInvariants?: boolean;
};

export function run<Func extends RunFuncType>(
  f: Func,
  options?: Omit<RunOptions<Func, never, never>, 'successActionCreator' | 'failActionCreator'>
): RunCmd;

export function run<
  Func extends RunFuncType,
  SuccessAction extends Action,
  >(
  f: Func,
  options: Omit<RunOptions<Func, SuccessAction, never>, 'successActionCreator'>
): RunCmd<SuccessAction, never>;

export function run<
  Func extends RunFuncType,
  FailAction extends Action,
  >(
  f: Func,
  options: Omit<RunOptions<Func, never, FailAction>, 'failActionCreator'>,
): RunCmd<never, FailAction>;

export function run<
  Func extends RunFuncType,
  SuccessAction extends Action,
  FailAction extends Action,
  >(
  f: Func,
  options: RunOptions<Func, SuccessAction, FailAction>,
): RunCmd<SuccessAction, FailAction>;
```

Reply:
I did something similar. Lemme know what you think.